content/js1/sprints/4/prep/index.md

```diff
@@ -8,4 +8,666 @@ backlog= 'Module-JS1'
+
+```json
+{
+  "name": "week-4-test-example",
+  "description": "An example application showing how to write tests using the jest framework",
+  "devDependencies": {
+    "jest": "^29.5.0"
+  }
+}
+```
+
+We've now got some additional information inside the `package.json`:
+
+```json
+"devDependencies" : "^29.5.0"
+```
+
+A **dependency** is a package that your project depends upon. Now that we're using Jest in our project, we **depend** on the Jest package. In particular, a **devDependency** is a **developer dependency** - a **dependency** we use for developing and testing our application.
+
+{{<note type="exercise" title="exercise">}}
+
+### 🔍 Figure it out
+
+There should be one more change to your project after running the installation command. Use documentation or ChatGPT to figure out what this change is for.
+{{</note>}}
+
+### 🕹️ Application Programming Interface
+
+With Jest installed, we need to figure out _how to use the Jest framework_ to write tests.
+Jest provides an **API** so we can use its functionality.
+
+API stands for
+
+- **Application**
+- **Programming**
+- **Interface**.
+
+We've encountered the word **interface** already.
+
+But we can break down each word in this acronym to understand it altogether.
+
+- An **application** is a program or piece of software designed to serve some purpose.
+- **Programming** refers to the process of writing code or software.
+- An 🕹️**interface** is a shared boundary across which two or more systems.
+
+{{<note type="Definition" title="Definition - API">}}
+
+An **API** is a boundary between a programmer and an application, enabling a programmer to use an application’s functionality without being concerned with how the application was built.
+
+We’ve encountered several functions like `console.log`, `Math.round` already.
+`console.log` and `Math.round` are **API**s.
+
+{{</note >}}
+
+{{<note type="exercise" title="exercise">}}
+
+Explain why `Math.round` is an **API**
+
+{{</note>}}
+
+{{<note type="exercise" title="exercise">}}
+
+### 🔍 Figure it out
+
+There should be one more change to your project after running the installation command. Use documentation or ChatGPT to figure out what this change is for.
+{{</note>}}
+
+### 🖌️ First test case
+
+> 🎯 Goal: Write the a test for the case below, using Jest:
+
+- Case 1 💼
+
+```js
+const input = 1;
+const currentOutput = getOrdinalNumber(input);
+const targetOutput = "1st";
+```
+
+To work out how to write a test using Jest, we can check out the documentation.
+
+We can create a file called `get-ordinal-number.js` and in there write our first test.
+We can figure out how to write this test using the Jest documentation. We can start writing a test:
+
+get-ordinal-number.test.js
+
+```js
+test("converts 1 to an ordinal number", function () {});
+```
+
+Let's break down this syntax.
+The `test` function is part of the Jest API: it is an interface we use to perform a particular task.
+In particular, we're using `test` to create a test case.
```

Comment:
```suggestion
In particular, we're using `test` to create a test case.
Before, we could use `Math.round` and `console.log` because `Math` and `console` are provided for us by Node. `test` isn't provided by Node, but when we ask Jest to run our tests, it will make sure the `test` function exists and that our code can use it.
```